R/sim_from_draws.R

```diff
@@ -50,14 +50,30 @@ sim_from_draws <- function(
   }
   
   parameter_names <- c(names(par_table$posterior), names(parameters))
-  if(! all(attr(model, "parameters") %in% parameter_names)) {
+  pkpdsim_parameter_names <- attr(model, "parameters")
+  if(length(grep("kappa_", pkpdsim_parameter_names)) > 0) { ## filter out IOV parameters
+    pkpdsim_parameter_names <- pkpdsim_parameter_names[-grep("kappa_", pkpdsim_parameter_names)]
+  }
+  if(! all(pkpdsim_parameter_names %in% parameter_names)) {
     stop(
       "Not all parameters are available in posterior draws or specified using ", 
       "the `parameters` argument. If parameter names are different in Stan vs ",
       "PKPDsim model, please use `map` argument to translate parameter names."
     )
   }
 
+  ## add IOV parameters, if not already specified
```

Comment:
Currently we don't support IOV in PKPDposterior, so we just keep them out of the implementation. But we need to have ability to transition to the same model in PKPDsim that has IOV parameters (`kappa_...`)